spark/src/main/scala/org/apache/spark/sql/comet/CometScanExec.scala

```diff
@@ -439,8 +440,29 @@ case class CometScanExec(
 
 object CometScanExec {
```

Comment:
perhaps we can also consider `CometScanExec extends TreeNode` to remove `mapProductIterator` method copy?

Reply:
We want to run `mapProductIterator` on `FileSourceScanExec`. I'm not sure how `object CometScanExec extends TreeNode` can help us?

Reply:
You are right, its needed to run `mapProductIterator` on `FileSourceScanExec`, I wrongly assumed this method needed for `CometScanExec `.